### PR TITLE
:recycle: Refactor exporting albums into archives

### LINKED ISSUE #28: Refactor Archive endpoint
* Move logic into a service
* Break logic down into multiple services
* Use Laravel HTTP response for binaries
* Add Unit tests

### PR DESCRIPTION
**Pull request by ltribolet**
Add rules and requests for download that suites better
laravel way.
Break down into different services classes the logic of
gathering photos by albums and compressing them into an
archive.

Fix #28

### CODE REVIEW

File: app/Http/Controllers/AlbumController.php [hunk 1]
```diff
@@ -348,7 +341,7 @@ public function setPublic(Request $request)
 
         if ($request->has('password')) {
             $password = $request->get('password', '');
-            $album->password = $password !== '' ? Hash::make($password) : null;
+            $album->password = empty($password) ? Hash::make($password) : null;
```
Comment (sanghin) [inline app/Http/Controllers/AlbumController.php:344]: If it's empty you want to return null ? 🤔

File: app/ControllerFunctions/ReadAccessFunctions.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ public function album(Album $album, bool $obeyHidden = false): int
             return 2;
         }
 
-        if ($album->password === '') {
+        if (empty($album->password)) {
```
Comment (sanghin) [inline app/ControllerFunctions/ReadAccessFunctions.php:58]: Check if he is not private bebore granting access?

File: app/Http/Controllers/AlbumController.php (hunk 1 above)
Comment (ltribolet) [reply app/Http/Controllers/AlbumController.php:344]: If it's empty then no password is set.

File: app/ControllerFunctions/ReadAccessFunctions.php (hunk 2 above)
Comment (ltribolet) [reply app/ControllerFunctions/ReadAccessFunctions.php:58]: It is done above L51

File: app/Http/Controllers/AlbumController.php (hunk 1 above)
Comment (ltribolet) [reply app/Http/Controllers/AlbumController.php:344]: The problem I got is that legacy is setting no password as an empty string, when I did some refactoring I changed it to null as it is better.

But there were many strict types checking with empty strings. Hence the issue you had when setting no password on the album but still getting a password in DB.

Comment (sanghin) [reply app/Http/Controllers/AlbumController.php:344]: But if there is a password you set it to null? I don't understand this condition :D

Comment (ltribolet) [reply app/Http/Controllers/AlbumController.php:344]: Ah I see my issue!

If it's empty I DO want to set it to null, not the other way :joy:

Comment (sanghin) [reply app/Http/Controllers/AlbumController.php]: This will fix #37 too ?

Comment (ltribolet) [reply app/Http/Controllers/AlbumController.php]: No I don't think so I've opened #37 when doing this and I'm not sure what's going on.

**Pull request merged**